src/Google.Datastore.V1Beta3/ArrayValuePartial.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Google.Type;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Collections;
+
+namespace Google.Datastore.V1Beta3
+{
+    // Additional helper members.
+    public partial class ArrayValue : IEnumerable<Value>
+    {
+        public static implicit operator ArrayValue(string[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(long[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(long?[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(Key[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(double[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(double?[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(bool[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(bool?[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(ArrayValue[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(ByteString[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(byte[][] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(Entity[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(LatLng[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(Timestamp[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(DateTime[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(DateTime?[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(DateTimeOffset[] values) => ToArrayValue(values, x => x);
+        public static implicit operator ArrayValue(DateTimeOffset?[] values) => ToArrayValue(values, x => x);
+
+        private static ArrayValue ToArrayValue<T>(T[] values, Func<T, Value> converter) =>
```

Comment:
(After offline discussion...) It's needed because each call needs the implicit conversion from `T` to `Value`... and there's no way to constrain `T` to support that.